server/src/main/java/org/elasticsearch/http/DefaultRestChannel.java

```diff
@@ -90,6 +90,7 @@ public void sendResponse(RestResponse restResponse) {
             if (content instanceof Releasable) {
                 toClose.add((Releasable) content);
             }
+            toClose.add(this::releaseOutputBuffer);
```

Comment:
This should really be right at the beginning of this method body, if we get here we already have the buffer all set up. If we do it later an e.g. the cors logic NPEs we leak memory.